streams/src/main/java/org/apache/kafka/streams/processor/internals/PartitionGroup.java

```
@@ -84,6 +84,17 @@ RecordQueue queue() {
         streamTime = RecordQueue.UNKNOWN;
     }
 
+    long getPartitionTimestamp(final TopicPartition partition) {
```

Should we check for partition existence for 2 calls?

---

What do you mean by "for 2 calls"? The call to `partitionTime()`? If the partition does not exists, it's a bug anyway and current code throws a NPE. What do we gain by checking if the partition exists? There is still a bug and we would need to throw an exception, too, IMHO. What do we gain by manually throwing an exception?

---

Nit: Can we rename this to `partitionTime` (to align the name to `streamTime()` method) -- in Kafka, we usually omit the `get` prefix on getter methods, and a record has a "timestamp" while for a partition or task it's a "time" (of course, both a "timestamp" and a "time" is just a long and quite similar, whoever, it seems more accurate to refer to their semantic meaning correctly).